runtime/commonMain/src/kotlinx/serialization/json/internal/JsonReader.kt

```diff
@@ -139,7 +139,27 @@ internal class JsonReader(private val source: String) {
     }
 
     fun takeString(): String {
-        if (tokenClass != TC_OTHER && tokenClass != TC_STRING) fail("Expected string or non-null literal", tokenPosition)
+        if (tokenClass != TC_OTHER && tokenClass != TC_STRING) fail(
+            "Expected string or non-null literal. $lenientHint",
```

Comment:
`takeString()` function is mainly called with `isLenient = true`, so '$lenientHint' is kinda misleading here.